### ISSUE TITLE
`absolute_paths` can trigger on code produced by a macro

### ISSUE THREAD

**Issue by ilyvion** [rust-lang/rust-clippy#13920]
### Summary

It looks like `absolute_paths` isn't properly "shielded" from macro code as something as simple as deriving `serde::Deserialize` on any type that holds another type with a generic sets it off.

### Lint Name

absolute_paths

### Reproducer

I tried this code:

```rust
use serde::Deserialize;

#[derive(Debug, Deserialize)]
struct Foo {
    bar: Vec<char>,
}
```

I saw this happen:

```
error: consider bringing this path into scope with the `use` keyword
 --> examples\min-repro.rs:9:5
  |
9 |     bar: Vec<char>,
  |     ^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absolute_paths
  = note: `-D clippy::absolute-paths` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(clippy::absolute_paths)]`

error: consider bringing this path into scope with the `use` keyword
 --> examples\min-repro.rs:9:5
  |
9 |     bar: Vec<char>,
  |     ^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absolute_paths
```

I expected to see this happen:

Nothing, obviously. There are no absolute paths in non-macro generated code here.

### Version

```text
rustc 1.85.0-nightly (d117b7f21 2024-12-31)
binary: rustc
commit-hash: d117b7f211835282b3b177dc64245fff0327c04c
commit-date: 2024-12-31
host: x86_64-pc-windows-msvc
release: 1.85.0-nightly
LLVM version: 19.1.6
```

**Comment by Centri3** [comment]
[It is.](https://github.com/rust-lang/rust-clippy/blob/c52740c87b70b576eb44ef4386ad49c59a9f30f3/clippy_lints/src/absolute_paths.rs#L111). I guess this is more procedural macros weirdness... Or this could very well be a bug in `Path`'s implementation of `WithSearchPat`.

I see that `TypeRelative` doesn't check the beginning of the span, which might be why this happens. I don't recall why I did this but I do remember it was intentional.

**Comment by ilyvion** [comment]
I don't know if it's the same cause, but since it's a similar problem, I'm wondering if this is a bigger problem than just this one lint.

I'm now also getting a similar problem with Bevy's `Reflect` macro:
```
error: this ident consists of a single char
  --> src\atlas_sprites\gizmos.rs:50:35
   |
50 |     #[reflect(Component, Default, Debug)]
   |                                   ^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#min_ident_chars
   = note: requested on the command line with `-D clippy::min-ident-chars`
```

Code:
```rust
use bevy::prelude::*;

#[derive(Component, Reflect, Default, Debug)]
#[reflect(Component, Default, Debug)]
pub struct ShowAtlasSpritesGizmos {
    pub render_text_anchor_point: Option<bool>,
    pub render_character_anchor_point: bool,
    pub render_character_box: bool,
}
```

**Comment by Centri3** [comment]
We don't seem to check for procedural macros in `min_ident_chars` if [it is from a pattern](https://github.com/rust-lang/rust-clippy/blob/b03370b9ce3b2cc0365779862f6c4756ad7de7e2/clippy_lints/src/min_ident_chars.rs#L88). I have a feeling `Debug` triggers it bc of its `f` argument. This might be an easy fix.

I'm still kinda unsure why the former happens (it's still probably the bug in `WithSearchPat`), but I'd recommend opening a separate issue for this one if this is still an issue

**Comment by ada4a** [comment]
both of those seem to be no longer reproducible! (tested on stable 1.89.0 and nightly 1.91.0 2025-07-30)

@rustbot label S-fixed

**Comment by sww1235** [comment]
Getting a similar issue with `thiserror`'s `#[from]` attribute. Since I am getting 2 reports about it when running `cargo clippy`, on one enum variant, I am assuming there is macro shenanigans going on.

**Comment by CommanderStorm** [comment]
since the issue seems no longer reproducible, I am closing it.

@sww1235 could you post the code you are seeing it on in another issue, so it can get properly adressed?

**Comment by sww1235** [comment]
> since the issue seems no longer reproducible, I am closing it.
> 
> [@sww1235](https://github.com/sww1235) could you post the code you are seeing it on in another issue, so it can get properly adressed?

Will test with the fixed code and file a new report if the issue persists.

**Comment by CommanderStorm** [comment]
Thanks 🙏
